src/nfa/nfa.cc

```diff
@@ -1267,20 +1257,50 @@ StateSet Nfa::post(const StateSet& states, const Symbol& symbol) const {
     return res;
 }
 
-void Mata::Nfa::uni(Nfa *unionAutomaton, const Nfa &lhs, const Nfa &rhs) {
-    *unionAutomaton = rhs;
+TransitionList::const_iterator Nfa::Nfa::get_epsilon_transitions(const State& state, const Symbol& epsilon) const {
+    assert(get_num_of_states() >= state + 1);
+    const auto trans_from = this->get_transitions_from(state);
```

Comment:
To not making a copy of the transitions.
```suggestion
    const auto& trans_from = this->get_transitions_from(state);
```